pandas/core/dtypes/cast.py

```diff
@@ -478,6 +478,42 @@ def maybe_cast_pointwise_result(
     return result
 
 
+def maybe_cast_to_pyarrow_dtype(
+    result: ArrayLike, converted_result: ArrayLike
+) -> ArrayLike:
+    """
+    Try casting result of a pointwise operation to its pyarrow dtype if
+    appropriate.
+
+    Parameters
+    ----------
+    result : array-like
+        Result to cast.
+
+    Returns
+    -------
+    result : array-like
+        result maybe casted to the dtype.
+    """
+    try:
+        import pyarrow as pa
+        from pyarrow import (
+            ArrowInvalid,
+            ArrowNotImplementedError,
+        )
+
+        from pandas.core.construction import array as pd_array
+
+        result[isna(result)] = np.nan
```

Comment:
Why is this necessary?

Reply:
because pyarrow library is not able to detect NA object when converting from numpy, it can only detect np.nan